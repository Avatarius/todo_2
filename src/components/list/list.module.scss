.list {
  display: flex;
  flex-direction: column;
  font-size: 30px;
  min-inline-size: 500px;
  max-inline-size: 402px;
  block-size: 550px;
  overflow-y: auto;
}

.list::-webkit-scrollbar {
  inline-size: 6px;
  margin-left: 31px;
}

.list::-webkit-scrollbar-thumb {
  background-color: rgb(82, 82, 82);
}

.item {
  padding: 7px;
  display: grid;
  grid-template-columns: min-content 1fr;
  align-items: center;
  gap: 15px;
  padding-inline-end: 35px;
  min-inline-size: 0;
  position: relative;
  border-block-end: 1px solid var(--border-color, #77777780);
}

.item:hover button[data-remove-button] {
  opacity: 1;
}

.text {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}
