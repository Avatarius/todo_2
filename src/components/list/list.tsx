import { ITodo } from "../../utils/types";
import { ButtonCheckmark } from "../buttonCheckmark/buttonCheckmark";
import { ButtonRemove } from "../buttonRemove/buttonRemove";
import styles from "./list.module.scss";

interface IListProps {
  todos: ITodo[];
  changeTodoStatus: (id: string) => void;
  removeTodo: (id: string) => void;
}

function List({ todos, changeTodoStatus, removeTodo }: IListProps) {
  return (
    <ul className={styles.list} data-testid='list'>
      {todos.map((todo) => {
        const {id, isCompleted} = todo;
        return (
          <li className={styles.item} key={id}>
            <ButtonCheckmark
              isComplated={isCompleted}
              onClick={() => changeTodoStatus(id)}
            />
            <p className={styles.text}>{todo.text}</p>
            <ButtonRemove onClick={() => removeTodo(id)} />
          </li>
        );
      })}
    </ul>
  );
}
export { List };
