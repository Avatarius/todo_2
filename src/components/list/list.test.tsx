import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { expect, test, describe, jest } from "@jest/globals";
import { List } from "./list";

describe("Компонент <List/>", () => {
  const todos = [
    { text: "test1", isCompleted: false, id: "1" },
    { text: "test2", isCompleted: true, id: "2" },
  ];
  test("Рендер компонента", () => {
    const component = renderer.create(
      <List
        todos={todos}
        changeTodoStatus={() => {}}
        removeTodo={(id: string) => {}}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Change todo status", () => {
    const changeTodoStatus = jest.fn();
    const listEl = render(
      <List
        todos={todos}
        changeTodoStatus={changeTodoStatus}
        removeTodo={() => {}}
      />
    );
    const list = screen.getByTestId("list");
    expect(list).toBeTruthy();
    const firstCheckmark = list.querySelector("[data-testid=buttonCheckmark]");
    expect(firstCheckmark).toBeTruthy();
    fireEvent.click(firstCheckmark as Element);
    expect(changeTodoStatus).toHaveBeenCalledWith("1");
  });
  test("Remove item", () => {
    const removeTodo = jest.fn();
    const listEl = render(
      <List todos={todos} changeTodoStatus={() => {}} removeTodo={removeTodo} />
    );
    const list = screen.getByTestId("list");
    expect(list).toBeTruthy();
    const firstRemoveButton = list.querySelector("[data-remove-button]");
    expect(firstRemoveButton).toBeTruthy();
    fireEvent.click(firstRemoveButton as Element);
    expect(removeTodo).toHaveBeenCalledWith("1");
  });
});
