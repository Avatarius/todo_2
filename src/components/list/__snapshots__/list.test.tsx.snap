// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <Input/> Рендер компонента 1`] = `
<ul
  data-testid="list"
>
  <li>
    <button
      data-testid="buttonCheckmark"
      onClick={[Function]}
      type="button"
    >
      <svg
        className=""
        fill="currentColor"
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.16 28a3 3 0 0 1-2.35-1.13l-6.59-8.25a1 1 0 0 1 1.56-1.24l6.59 8.24a1 1 0 0 0 1.63-.06L27.17 4.44a1 1 0 1 1 1.66 1.12L14.67 26.67A3 3 0 0 1 12.29 28Z"
        />
      </svg>
    </button>
    <p>
      test1
    </p>
    <button
      data-remove-button={true}
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="200"
        viewBox="0 0 490 490"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M456.851 0 245 212.564 33.149 0 .708 32.337l211.961 212.667L.708 457.678 33.149 490 245 277.443 456.851 490l32.441-32.322-211.961-212.674L489.292 32.337z"
        />
      </svg>
    </button>
  </li>
  <li>
    <button
      data-testid="buttonCheckmark"
      onClick={[Function]}
      type="button"
    >
      <svg
        className=""
        fill="currentColor"
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.16 28a3 3 0 0 1-2.35-1.13l-6.59-8.25a1 1 0 0 1 1.56-1.24l6.59 8.24a1 1 0 0 0 1.63-.06L27.17 4.44a1 1 0 1 1 1.66 1.12L14.67 26.67A3 3 0 0 1 12.29 28Z"
        />
      </svg>
    </button>
    <p>
      test2
    </p>
    <button
      data-remove-button={true}
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="200"
        viewBox="0 0 490 490"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M456.851 0 245 212.564 33.149 0 .708 32.337l211.961 212.667L.708 457.678 33.149 490 245 277.443 456.851 490l32.441-32.322-211.961-212.674L489.292 32.337z"
        />
      </svg>
    </button>
  </li>
</ul>
`;
